Code/Framework/AzCore/AzCore/Math/Internal/SimdMathVec3_neon.inl

```diff
@@ -353,26 +353,38 @@ namespace AZ
 
         AZ_MATH_INLINE bool Vec3::CmpAllEq(FloatArgType arg1, FloatArgType arg2)
         {
+            // NeonQuad::CmpAllEq compares all 4 elements.
+            // The fourth elements are already equal since it's not used in Vec3.
             return NeonQuad::CmpAllEq(arg1, arg2);
         }
 
         AZ_MATH_INLINE bool Vec3::CmpAllLt(FloatArgType arg1, FloatArgType arg2)
         {
-            return NeonQuad::CmpAllLt(arg1, arg2);
```

Comment:
It's annoying Neon doesn't seem to have masked instructions. But I wonder if it would be better to add a mask to the signature for NeonQuad::CmpAllLt and the other functions, and then it can be fixed at a lower level rather than at the call site.

Reply:
By looking at the implementation of all the `NeonQuad::CmpAll` functions, they rely on `Neon::AreAllLanesTrue` which simply does `vminvq_u32(value) != 0;` (it just takes the minimum of all the elements and returns it, if the minimum is not 0 then all of them must be true). 

Since that is a unique way of handling this, I thought it was cleaner just to manage the fourth element before calling the function in Vec3, rather than changing the functions of `NeonQuad::CmpAll` to support any mask. The change is also limited to Vec3 internal functions, it doesn't spread beyond that.

Maybe we can submit this fix now and create a task to modify `NeonQuad::CmpAll` in the future to support masks. What do you think?

Reply:
We chatted about this on slack and I think it's probably fine to fix this here rather than at a lower level, because Vector3 will be the only place using the Quad types with a dummy fourth element.

Will the fourth element always be zero? I'm a bit worried some of the other operations might allow different values to wind up in there.

Reply:
>Will the fourth element always be zero? I'm a bit worried some of the other operations might allow different values to wind up in there.

Yeah, that's a valid concern, I haven't looked at all the other operations to see if fourth element could be randomly affected.

I guess it'd be safer to read the value of one element and assign it to the other with a small increment. It's more operations, but I guess it's worth it, otherwise it could be a very difficult bug to find because it might start failing after certain operations only. Let me do that modification now.

Reply:
@greerdv I found a way to do it with masks, but as I was changing it I realized that is a rather larger change and since at the moment I can't run unit tests I think it'll be safer to submit it with the manual modification of the fourth element. I've changed the code so now both arg1 and arg2 fourth elements are set to a valid that the comparison operation will return true.

Once I'm able to run unit tests on the phone I'll change the implementation to add masks.

Reply:
Sounds good. Best to keep the change as small as possible until we have the tests running to validate.

Reply:
The replace 4th with scalar instruction basically (or should) compiles down to a single VMOV.32 instruction. From what I can find all ARM CPU's should issue this in 1 cycle and complete within 4. My guess is on all modern CPU's we won't notice that hit, but we have all the benchmarking in place to prove this much better than my intuition can.

Changes look good, thanks!